fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -2066,12 +2120,17 @@ public void finishTransactionBatch(TransactionStateBatch stateBatch, Set<Long> e
             try {
                 writeLock();
                 try {
+                    // check whether version is consistent
+                    if (!isTxnStateBatchConsistent(db, stateBatch)) {
+                        return;
+                    }
+
```

Comment:
better check before `writeLock();` and `stateBatch.afterVisible()` after `writeUnlock()`?